soccer/src/soccer/planning/primitives/create_path.cpp

```diff
@@ -75,4 +75,102 @@ Trajectory rrt(const LinearMotionInstant& start, const LinearMotionInstant& goal
+                        const MotionConstraints& motion_constraints, RJ::Time start_time,
+                        const rj_geometry::ShapeSet& static_obstacles) {
+    // if already on goal, no need to move
+    if (start.position.dist_to(goal.position) < 1e-6) {
+        return Trajectory{{RobotInstant{Pose(start.position, 0), Twist(), start_time}}};
+    }
+
+    // maybe straight line works
+    Trajectory straight_trajectory =
+        CreatePath::simple(start, goal, motion_constraints, start_time);
+
+    // If we are very close to the goal (i.e. there physically can't be a robot
+    // in our way) or the straight trajectory is feasible, we can use it.
+    if (start.position.dist_to(goal.position) < kRobotRadius ||
+        (!trajectory_hits_static(straight_trajectory, static_obstacles, start_time, nullptr))) {
+        return straight_trajectory;
+    }
+
+    // Generate list of intermediate points
+    std::vector<rj_geometry::Point> intermediates = get_intermediates(start, goal);
+
+    for (int i = 0; i < intermediate::PARAM_num_intermediates; i++) {
+        rj_geometry::Point final_inter = intermediates[i];
+
+        // Step through the path from the robot to the final intermediate point
+        // and test each point on that path as an intermediate point
+    for (double t = intermediate::PARAM_step_size; t < final_inter.dist_to(start.position); t += intermediate::PARAM_step_size) {
+            rj_geometry::Point intermediate =
+                (final_inter - start.position).normalized(t) + start.position;
+            Trajectory trajectory =
+                CreatePath::simple(start, goal, motion_constraints, start_time, {intermediate});
+
+            // If the trajectory does not hit an obstacle, it is valid
+            if ((!trajectory_hits_static(trajectory, static_obstacles, start_time, nullptr))) {
+                cached_intermediate_pair_ = {(final_inter - start.position).angle(), (final_inter - start.position).mag()};
+                return trajectory;
+            }
+        }
+    }
+
+    // If all else fails, return the straight-line trajectory
+    return straight_trajectory;
+}
+
+std::vector<rj_geometry::Point> get_intermediates(const LinearMotionInstant& start,
+                                                  const LinearMotionInstant& goal) {
+    std::random_device rd;
+    std::mt19937 gen(rd());
+    // Create a random distribution for the distance between the start
+    // and the intermediate points
+    std::uniform_real_distribution<> scale_dist(intermediate::PARAM_min_scale, intermediate::PARAM_max_scale);
+
+    double angle_range = intermediate::PARAM_max_angle - intermediate::PARAM_min_angle;
+    // Create a random distribution for the angle between the start
+    // and the intermediate points
+    std::uniform_real_distribution<> angle_dist(-angle_range, angle_range);
+
+    std::vector<rj_geometry::Point> intermediates;
+    std::vector<std::pair<double, double>> inter_pairs;
```

Comment:
Instead, use std::tuple<double, double, double> where the first value is abs(angle) second angle, third scale. This allows sort to work correctly since it compares the first value of the tuple